### PR TITLE
Create csv

### PR DESCRIPTION
**Pull request by JackPlowman**
Added functionality 
-Choice of custom path
-Custom Paths

Have Not added Yet
-Test Cases (coming ASAP)

### CODE REVIEW

File: street_level_crime/create_csv.py [hunk 1]
```diff
@@ -7,13 +9,52 @@ def convert_to_csv(headers: tuple, data: list) -> csv:
     :Tuple headers: Contians the headers of the csv file
     :List data: Contains a list of row data in format List[List[String]]
     """
+    print("When entering a filename please don't add .csv to the file")
+    filename = input("Please enter a filename: ")
```
Comment (eperegrine) [inline street_level_crime/create_csv.py]: Use the ui_helpers.py methods to get input

Comment (eperegrine) [reply street_level_crime/create_csv.py]: Explain what the file path is used for

File: street_level_crime/create_csv.py [hunk 2]
```diff
@@ -7,13 +9,52 @@ def convert_to_csv(headers: tuple, data: list) -> csv:
...
-            csvWriter = csv.writer(f)  # creates writer object
-            csvWriter.writerow(headers)  # writes header
-            for i in range(0, len(data) - 1, 1):  # writes each row in array
-                csvWriter.writerow(data[i])  # writes from
-
+        with open(filename + '.csv', 'w+', newline='') as f:  # creates document and opens for writer object
+            csvWriter = csv.writer(f)
+            csvWriter.writerow(headers)
+            for i in range(0, len(data) - 1, 1):
+                csvWriter.writerow(data[i])
```
Comment (eperegrine) [inline street_level_crime/create_csv.py]: No need to use len
```suggestion
            for row in data:
                csvWriter.writerow(row)
```

File: street_level_crime/create_csv.py [hunk 3]
```diff
@@ -7,13 +9,52 @@ def convert_to_csv(headers: tuple, data: list) -> csv:
...
+            print("Do you want to save to a custom path?")
+            print("[1] Yes")
+            print("[2] No")
+            CustomPathInput = input()
+            special_commands(CustomPathInput)
+            if CustomPathInput == "1":
+                custom_path(filename)
+            else:
+                print("Saved to default location: " + os.path.dirname(
+                    os.path.realpath(filename + ".csv")) + '\\' + filename + ".csv")
```
Comment (eperegrine) [inline street_level_crime/create_csv.py]: Superfluous - the first filename input should specify the path as well if desired

File: street_level_crime/create_csv.py [hunk 4]
```diff
@@ -7,13 +9,52 @@ def convert_to_csv(headers: tuple, data: list) -> csv:
...
+            print("Saved to: " + dst)
+        else:
+            print("Saved to default location: " + src)
+    except:
+        print("Unable to save file to custom path")
+
+
+def special_commands(variable):
+    pass
+    # Link to Jack Gittoes' Code
```
Comment (eperegrine) [inline street_level_crime/create_csv.py]: The commands stuff is a tricky one, Need to have a look at what jack has done with the UI but we'll find a unified way of doing this

File: street_level_crime/test.py [hunk 5]
```diff
@@ -69,3 +71,11 @@ def test_process_line_skip_none(self):
 
 if __name__ == '__main__':
     unittest.main()
+
+
+class TestConvertToCSV(unittest.TestCase):
+    def test_convert_to_csv(self):
+        headers = ['a', 'b', 'c']
+        data = [['a', 'b', 'c'], ['d', 'e', 'f'], ['g', 'h', 'i'], ['j', 'k', 'l'], ['m', 'n', 'o']]
+        with mock.patch('builtins.input', return_value="Jack"):
```
Comment (eperegrine) [inline street_level_crime/test.py]: Testing file writing is a tricky one, would likely need to be a manual check

**Review by eperegrine — CHANGES_REQUESTED** [review]

File: street_level_crime/create_csv.py (hunk 1 above)
Comment (eperegrine) [reply street_level_crime/create_csv.py]: This method should take a path as a parameter, instead of requesting input

Comment (JackPlowman) [reply street_level_crime/create_csv.py]: Done

File: street_level_crime/test.py (hunk 5 above)
Comment (JackPlowman) [reply street_level_crime/test.py]: Removed for now

File: street_level_crime/create_csv.py (hunk 4 above)
Comment (JackPlowman) [reply street_level_crime/create_csv.py]: Waiting for Jack's new functions

**Pull request merged**